docs/design/services/dynamodb/high-level-library/prototype/option-1/sync/Prototype.java

```diff
@@ -0,0 +1,1025 @@
+     *                                        .build();
+     *
+     *         client.createTable(CreateTableRequest.builder()
+     *                                              .tableName("my-table")
+     *                                              .provisionedCapacity(tableCapacity)
+     *                                              .keySchema(tableKeys)
+     *                                              .build());
+     *
+     *         System.out.println("Table created successfully.");
+     *     } catch (TableAlreadyExistsException e) {
+     *         System.out.println("Table creation failed.");
+     *     }
+     * </code>
+     */
+    CreateTableResponse createTable(CreateTableRequest createTableRequest)
+            throws TableAlreadyExistsException;
+
+    /**
+     * Get a specific DynamoDB table, based on its table name. If the table does not exist, use {@link #createTable}.
+     *
+     * Usage Example:
+     * <code>
+     *     try (DynamoDbDocumentClient client = DynamoDb.documentClient()) {
+     *         Table table = client.getTable("my-table");
+     *         System.out.println(table);
+     *     } catch (NoSuchTableException e) {
+     *         System.out.println("Table does not exist.");
+     *     }
+     * </code>
+     */
+    Table getTable(String tableName)
+            throws NoSuchTableException;
+
+    /**
+     * Get a lazily-populated iterable over all DynamoDB tables on the current account and region.
+     *
+     * Usage Example:
+     * <code>
+     *     try (DynamoDbDocumentClient client = DynamoDb.documentClient()) {
+     *         String tables = client.listTables().tables().stream()
+     *                               .map(Table::name)
+     *                               .collect(Collectors.joining(","));
+     *         System.out.println("Current Tables: " + tables);
+     *     }
+     * </code>
+     */
+    ListTablesResponse listTables();
+
+    /**
+     * The builder for the high-level DynamoDB client. This is used by customers to configure the high-level client with default
+     * values to be applied across all client operations.
+     *
+     * This can be created via {@link DynamoDb#documentClientBuilder()} or {@link DynamoDbDocumentClient#builder()}.
+     */
+    interface Builder {
+        /**
+         * Configure the DynamoDB document client with a low-level DynamoDB client.
+         *
+         * Default: {@code DynamoDbClient.create()}
+         */
+        DynamoDbDocumentClient.Builder dynamoDbClient(DynamoDbClient client);
+
+        /**
+         * Configure the DynamoDB document client with a specific set of configuration values that override the defaults.
+         *
+         * Default: {@code DocumentClientConfiguration.create()}
+         */
+        DynamoDbDocumentClient.Builder documentClientConfiguration(DocumentClientConfiguration configuration);
```

Comment:
I wonder if we should unify this configuration builder method name for all HLLs to make it more discoverable,  overrideConfiguraion?

Reply:
The cons of that is that we will deviate from the type-matches-the-setter-name approach we have elsewhere.

We could call it DocumentOverrideConfiguration or something similarly long so that we can have our cake and eat it too.